hefquin-engine/src/main/java/se/liu/ida/hefquin/engine/federation/access/impl/CardinalityCacheMap.java

```diff
@@ -0,0 +1,213 @@
+package se.liu.ida.hefquin.engine.federation.access.impl;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.util.Collection;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
+
+import se.liu.ida.hefquin.engine.federation.FederationMember;
+import se.liu.ida.hefquin.engine.federation.access.CardinalityResponse;
+import se.liu.ida.hefquin.engine.federation.access.DataRetrievalRequest;
+
+/**
+ * A thread-safe cache implementation for storing cardinality responses. This
+ * cache uses a ConcurrentHashMap and supports serialization to persist data.
+ *
+ * @param <K> The key type for caching cardinality responses.
+ */
+public class CardinalityCacheMap<K> implements Map<K, CompletableFuture<CardinalityResponse>> {
```

Comment:
Instead of just having this class being an implementation of `Map`, please introduce a `PersistableCache` sub-interface of our `Cache` interface and, then, make this class an implementation of this sub-interface.

Reply:
[5a30aa6](https://github.com/LiUSemWeb/HeFQUIN/commit/5a30aa61bac120696fac6767cad2f44fca90b6a4)